test_suite/gap/test_gap_privacy.py

```diff
@@ -73,640 +31,347 @@ def address_is_random_non_resolvable(address):
-    # disconnected
-    sleep(1)
+    central_ss.connect(peripheral_ss)
 
+    # Always start peripheral_ss first
+    peripheral_ss.wait_for_event()
+    central_ss.start_pairing()
 
-@pytest.fixture(scope="function")
-def rand_data():
-    rand_data = get_rand_data("MANUFACTURER_SPECIFIC_DATA")
-    # make sure that the data have an acceptable size
-    if len(rand_data) > 16:
-        rand_data = rand_data[0:16]
-    return rand_data
+    # We should get a request
+    peripheral_ss.expect_pairing_request()
 
+    # Accept request, pairing should complete successfully
+    peripheral_ss.accept_pairing_request()
+    peripheral_ss.expect_pairing_success()
 
-@pytest.fixture(scope="function")
-def peripheral(board_allocator: BoardAllocator, rand_data):
-    device = board_allocator.allocate('peripheral')
-    assert device is not None
-    init_device(device)
+    # central_ss should see pairing complete successfully too
+    central_ss.expect_pairing_success()
 
-    configure_peripheral(device, rand_data)
-    yield device
+    central.gap.disconnect(peripheral_ss.connection_handle, "USER_TERMINATION")
 
-    device.ble.shutdown()
-    board_allocator.release(device)
+    # FIXME: needs event from the peripheral and central that they have been disconnected
+    sleep(0.5)
 
 
-@pytest.fixture(scope="function")
-def unknown_peripheral(board_allocator: BoardAllocator, rand_data):
-    device = board_allocator.allocate('peripheral')
-    assert device is not None
-    init_device(device)
-
-    configure_peripheral(device, rand_data)
-    yield device
-
-    device.ble.shutdown()
-    board_allocator.release(device)
-
-
-@pytest.fixture(scope="function")
-def central(board_allocator: BoardAllocator):
-    device = board_allocator.allocate('central')
-    assert device is not None
-    init_device(device)
-    device.gap.setCentralPrivacyConfiguration(
-        False,
-        "RESOLVE_AND_FORWARD"
-    )
-
-    yield device
-
-    device.ble.shutdown()
-    board_allocator.release(device)
-
-
-@pytest.fixture(scope="function")
-def unknown_central(board_allocator: BoardAllocator):
-    device = board_allocator.allocate('central')
-    assert device is not None
-    init_device(device)
-
-    yield device
-
-    device.ble.shutdown()
-    board_allocator.release(device)
-
-
-@pytest.fixture(scope="function")
-def peripheral_address(peripheral):
-    yield peripheral.gap.getAddress().result
+def init_privacy(device):
+    device.securityManager.init(True, False, "IO_CAPS_NONE", "*", True, "*")
+    device.gap.enablePrivacy(True)
 
 
 @pytest.mark.ble42
-@pytest.mark.skip(reason="Missing BLE API")
-@pytest.mark.parametrize("advertising_type", [
-     "ADV_CONNECTABLE_UNDIRECTED",
-     # "ADV_CONNECTABLE_DIRECTED",   # Directed advertising not supported yet
-     "ADV_SCANNABLE_UNDIRECTED",
-     "ADV_NON_CONNECTABLE_UNDIRECTED"
+@pytest.mark.parametrize("advertising_type, unresolvable", [
```

Comment:
I think we should remove the ble42 marker of these tests.

Reply:
changed to ble41